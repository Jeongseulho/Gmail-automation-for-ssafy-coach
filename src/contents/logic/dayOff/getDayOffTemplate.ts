import { DayOffCategory } from '@/types/DayOffCategory';
import { formatDate } from '@/utils/formatDate';

export const getDayOffTemplate = (cohort: string, date: string, campus: string, classGroup: string, name: string, dayOffCategory: DayOffCategory) => {
  const title = `[SSAFY] ${formatDate(date)} ${campus} ${classGroup} 실습코치 ${name} ${dayOffCategory} 신청서 송부`;
  const content = `
    <div>안녕하십니까, ${cohort} ${campus} 실습코치 ${name}입니다.</div>
    <br>
    <div>${formatDate(date)} ${dayOffCategory} 신청서 첨부파일로 송부 드립니다.</div>
    <br>
    <b>특이사항</b>
    <br>
    <div>- 해당 사항 없음.</div>
    <br>
    <div>감사합니다.</div>
    <div>${name} 드림.</div>
  `;
  return {
    title,
    content,
  };
};
