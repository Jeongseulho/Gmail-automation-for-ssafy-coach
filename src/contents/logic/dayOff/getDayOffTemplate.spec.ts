import { getDayOffTemplate } from './getDayOffTemplate';

describe('getDayOffTemplate 함수', () => {
  it('오전 반차 휴가 신청서 템플릿을 반환한다', () => {
    const { title, content } = getDayOffTemplate('10기', '240202', '구미', '이희진', '오전 반차');
    expect(title).toMatchSnapshot();
    expect(content).toMatchSnapshot();
  });
  it('오후 반차 휴가 신청서 템플릿을 반환한다', () => {
    const { title, content } = getDayOffTemplate('10기', '240202', '구미', '이희진', '오후 반차');
    expect(title).toMatchSnapshot();
    expect(content).toMatchSnapshot();
  });

  it('하루 휴가 신청서 템플릿을 반환한다', () => {
    const { title, content } = getDayOffTemplate('10기', '240202', '구미', '이희진', '하루 휴가');
    expect(title).toMatchSnapshot();
    expect(content).toMatchSnapshot();
  });
});
