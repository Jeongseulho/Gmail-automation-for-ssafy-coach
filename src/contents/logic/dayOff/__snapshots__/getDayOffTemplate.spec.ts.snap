// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDayOffTemplate 함수 오전 반차 휴가 신청서 템플릿을 반환한다 1`] = `"[SSAFY] 2월 2일(금) 구미 1반 실습코치 이희진 오전 반차 신청서 송부"`;

exports[`getDayOffTemplate 함수 오전 반차 휴가 신청서 템플릿을 반환한다 2`] = `
"
    <div>안녕하십니까, 10기 구미 실습코치 이희진입니다.</div>
    <br>
    <div>2월 2일(금) 오전 반차 신청서 첨부파일로 송부 드립니다.</div>
    <br>
    <b>특이사항</b>
    <br>
    <div>- 해당 사항 없음.</div>
    <br>
    <div>감사합니다.</div>
    <div>이희진 드림.</div>
  "
`;

exports[`getDayOffTemplate 함수 오후 반차 휴가 신청서 템플릿을 반환한다 1`] = `"[SSAFY] 2월 2일(금) 구미 1반 실습코치 이희진 오후 반차 신청서 송부"`;

exports[`getDayOffTemplate 함수 오후 반차 휴가 신청서 템플릿을 반환한다 2`] = `
"
    <div>안녕하십니까, 10기 구미 실습코치 이희진입니다.</div>
    <br>
    <div>2월 2일(금) 오후 반차 신청서 첨부파일로 송부 드립니다.</div>
    <br>
    <b>특이사항</b>
    <br>
    <div>- 해당 사항 없음.</div>
    <br>
    <div>감사합니다.</div>
    <div>이희진 드림.</div>
  "
`;

exports[`getDayOffTemplate 함수 하루 휴가 신청서 템플릿을 반환한다 1`] = `"[SSAFY] 2월 2일(금) 구미 1반 실습코치 이희진 휴가 신청서 송부"`;

exports[`getDayOffTemplate 함수 하루 휴가 신청서 템플릿을 반환한다 2`] = `
"
    <div>안녕하십니까, 10기 구미 실습코치 이희진입니다.</div>
    <br>
    <div>2월 2일(금) 휴가 신청서 첨부파일로 송부 드립니다.</div>
    <br>
    <b>특이사항</b>
    <br>
    <div>- 해당 사항 없음.</div>
    <br>
    <div>감사합니다.</div>
    <div>이희진 드림.</div>
  "
`;
